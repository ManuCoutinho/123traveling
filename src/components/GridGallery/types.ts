import { GalleryItemsProps } from 'components/GalleryItems/types'

export type GridGalleryProps = {
  items?: GalleryItemsProps[]
}
