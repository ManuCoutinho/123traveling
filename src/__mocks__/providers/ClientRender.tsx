/* eslint-disable react/display-name */

import { render } from '@testing-library/react'
import { QueryClient, QueryClientProvider } from 'react-query'
import { ChakraProvider } from '@chakra-ui/react'

const createTestQueryClient = () =>
	new QueryClient({
		defaultOptions: {
			queries: {
				retry: false
			}
		}
	})

export function renderWithClient(children: React.ReactNode) {
	const testQueryClient = createTestQueryClient()
	const { rerender, ...result } = render(
		<ChakraProvider>
			<QueryClientProvider client={testQueryClient}>
				{children}
			</QueryClientProvider>
		</ChakraProvider>
	)
	return {
		...result,
		rerender: (rerenderUi: React.ReactNode) =>
			rerender(
				<QueryClientProvider client={testQueryClient}>
					{rerenderUi}
				</QueryClientProvider>
			)
	}
}

export const customWrapper = () => {
	const testQueryClient = createTestQueryClient()

	return ({ children }: { children: React.ReactNode }) => (
		<QueryClientProvider client={testQueryClient}>
			{children}
		</QueryClientProvider>
	)
}
