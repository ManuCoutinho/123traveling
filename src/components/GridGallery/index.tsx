import { Box, Grid, Heading } from '@chakra-ui/react'
import { GalleryItems } from 'components/GalleryItems'
import { v4 as uuidv4 } from 'uuid'
import { GridGalleryProps } from './types'

const GridGallery: React.FC<GridGalleryProps> = ({ items }) => {
  return (
    <Box as='section' w={['300', '700', '1024']} p={[4, 6, 12, 16]} mx='auto'>
      <Heading color='gray.700' fontWeight='semibold' fontSize='4xl' lineHeight='taller' mb={6}>
        Cidades +100
      </Heading>
      <Grid templateColumns='repeat(auto-fit, minmax(250px, 1fr))' gap={[6, 8]}>
        {items?.map((item) => {
          return <GalleryItems {...item} key={uuidv4()} />
        })}
      </Grid>
    </Box>
  )
}
export default GridGallery
