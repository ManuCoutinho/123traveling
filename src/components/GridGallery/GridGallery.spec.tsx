import { screen } from '@testing-library/react'
import { renderWithClient } from '__mocks__/providers/ClientRender'
import GridGallery from '.'

const itemsMock = [
	{
		image: 'nxkinK7hhOk',
		city: 'Rio de Janeiro',
		country: 'Brasel',
		flag: 'rj'
	},
	{
		image: 'jisfldjj',
		city: 'Cairo',
		country: 'Egito',
		flag: 'eg'
	}
]
describe('<GridGallery/>', () => {
	it('should render GridGallery correctly ', () => {
		renderWithClient(<GridGallery items={itemsMock} />)
		expect(
			screen.getByRole('heading', { name: /cidades/i })
		).toBeInTheDocument()

		expect(
			screen.getByRole('img', { name: itemsMock[0].city })
		).toBeInTheDocument()
		expect(
			screen.getByRole('img', { name: itemsMock[0].country })
		).toHaveAttribute('src', 'https://flagcdn.com/w640/rj.png')
	})
	it('should match to snapshot', () => {
		const { container } = renderWithClient(
			<GridGallery items={itemsMock} />
		)
		expect(container).toMatchSnapshot()
	})
})
