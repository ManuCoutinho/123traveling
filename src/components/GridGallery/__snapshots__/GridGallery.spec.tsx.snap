// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridGallery/> should match to snapshot 1`] = `
<div>
  <section
    class="css-1d5e0cu"
  >
    <h2
      class="chakra-heading css-12f2yk3"
    >
      Cidades +100
    </h2>
    <div
      class="css-k303k8"
    >
      <div
        class="chakra-stack css-vbssdh"
      >
        <div
          class="css-0"
        >
          <a
            href="https://trips.manucoutinho.com/place/rio-de-janeiro-rj"
            rel="noopener noreferrer"
          >
            <img
              alt="Rio de Janeiro"
              data-nimg="1"
              decoding="async"
              height="400"
              loading="lazy"
              src=""
              style="color: transparent; width: 350px; height: 250px; object-fit: cover; border-radius: 6px;"
              width="300"
            />
          </a>
          <div
            class="chakra-stack css-1b6hh2u"
          >
            <p
              class="chakra-text css-18hwcuh"
            >
              Photo by
              <a
                class="chakra-link css-spn4bz"
                href="undefined?utm_source=excursionista&utm_medium=referral"
                rel="noopener"
                target="_blank"
              />
              on
              <a
                class="chakra-link css-spn4bz"
                href="https://unsplash.com/?utm_source=excursionista&utm_medium=referral"
                rel="noopener"
                target="_blank"
              >
                Unsplash
              </a>
            </p>
            <button
              aria-label="download image"
              class="chakra-button css-pe7a1m"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                focusable="false"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M.5 9.9a.5.5 0 0 1 .5.5v2.5a1 1 0 0 0 1 1h12a1 1 0 0 0 1-1v-2.5a.5.5 0 0 1 1 0v2.5a2 2 0 0 1-2 2H2a2 2 0 0 1-2-2v-2.5a.5.5 0 0 1 .5-.5z"
                />
                <path
                  d="M7.646 11.854a.5.5 0 0 0 .708 0l3-3a.5.5 0 0 0-.708-.708L8.5 10.293V1.5a.5.5 0 0 0-1 0v8.793L5.354 8.146a.5.5 0 1 0-.708.708l3 3z"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="css-8um7m7"
        >
          <div
            class="css-0"
          >
            <p
              class="chakra-text css-1n3b5wr"
            >
              Rio de Janeiro
            </p>
            <p
              class="chakra-text css-1jh104j"
            >
              Brasel
            </p>
          </div>
          <img
            alt="Brasel"
            crossorigin="anonymous"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Fflagcdn.com%2Fw640%2Frj.png&w=96&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fflagcdn.com%2Fw640%2Frj.png&w=48&q=75 1x, /_next/image?url=https%3A%2F%2Fflagcdn.com%2Fw640%2Frj.png&w=96&q=75 2x"
            style="color: transparent; border-radius: 100%; width: 50px; object-fit: cover; box-shadow: inset 0px 14px 15px -3px rgba(0,0,0,0.1),0px 10px 15px -3px rgba(0,0,0,0.1);"
            width="40"
          />
        </div>
      </div>
      <div
        class="chakra-stack css-vbssdh"
      >
        <div
          class="css-0"
        >
          <a
            href="https://trips.manucoutinho.com/place/cairo-eg"
            rel="noopener noreferrer"
          >
            <img
              alt="Cairo"
              data-nimg="1"
              decoding="async"
              height="400"
              loading="lazy"
              src=""
              style="color: transparent; width: 350px; height: 250px; object-fit: cover; border-radius: 6px;"
              width="300"
            />
          </a>
          <div
            class="chakra-stack css-1b6hh2u"
          >
            <p
              class="chakra-text css-18hwcuh"
            >
              Photo by
              <a
                class="chakra-link css-spn4bz"
                href="undefined?utm_source=excursionista&utm_medium=referral"
                rel="noopener"
                target="_blank"
              />
              on
              <a
                class="chakra-link css-spn4bz"
                href="https://unsplash.com/?utm_source=excursionista&utm_medium=referral"
                rel="noopener"
                target="_blank"
              >
                Unsplash
              </a>
            </p>
            <button
              aria-label="download image"
              class="chakra-button css-pe7a1m"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                focusable="false"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M.5 9.9a.5.5 0 0 1 .5.5v2.5a1 1 0 0 0 1 1h12a1 1 0 0 0 1-1v-2.5a.5.5 0 0 1 1 0v2.5a2 2 0 0 1-2 2H2a2 2 0 0 1-2-2v-2.5a.5.5 0 0 1 .5-.5z"
                />
                <path
                  d="M7.646 11.854a.5.5 0 0 0 .708 0l3-3a.5.5 0 0 0-.708-.708L8.5 10.293V1.5a.5.5 0 0 0-1 0v8.793L5.354 8.146a.5.5 0 1 0-.708.708l3 3z"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="css-8um7m7"
        >
          <div
            class="css-0"
          >
            <p
              class="chakra-text css-1n3b5wr"
            >
              Cairo
            </p>
            <p
              class="chakra-text css-1jh104j"
            >
              Egito
            </p>
          </div>
          <img
            alt="Egito"
            crossorigin="anonymous"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Fflagcdn.com%2Fw640%2Feg.png&w=96&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fflagcdn.com%2Fw640%2Feg.png&w=48&q=75 1x, /_next/image?url=https%3A%2F%2Fflagcdn.com%2Fw640%2Feg.png&w=96&q=75 2x"
            style="color: transparent; border-radius: 100%; width: 50px; object-fit: cover; box-shadow: inset 0px 14px 15px -3px rgba(0,0,0,0.1),0px 10px 15px -3px rgba(0,0,0,0.1);"
            width="40"
          />
        </div>
      </div>
    </div>
  </section>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
